matchmaking/tasks.py

```diff
@@ -0,0 +1,109 @@
+from datetime import timedelta, timezone, datetime
+import json
+import logging
+
+from background_task.signals import task_failed
+from background_task.models import Task
+from sentry_sdk import capture_message, capture_exception
+
+from common.utils import background
+from matchmaking.common import request_matches, make_matches, remind_matches
+from matchmaking.matchmaker import MatchMaker
+
+PERIOD_IN_DAYS = 15
+DAYS_AFTER_REQUEST_TO_MAKE_MATCH = 7
+DAYS_AFTER_MATCH_TO_REMIND = 7
+
+
+def periodically_request_matches(slack_team_id):
+    """first check if we already have a task for this slack team"""
+    # this is a little lazy way to implement this, and it ties our implementation to background_tasks, but
+    # I'd rather not think through and create a whole new model for SlackTeamsWithAutomatedMatchMaking just yet.
+    existing_request_match_tasks = Task.objects.filter(
+        task_name=_request_matches_task_name(),
+        task_params__contains=f'"slack_team_id": "{slack_team_id}"',
+    )
+    if len(existing_request_match_tasks) >= 1:
+        if len(existing_request_match_tasks) > 1:
+            capture_message(
+                'multiple _request_match_tasks for slack team',
+                extras={'slack_team_id': slack_team_id},
+            )
+            for task in existing_request_match_tasks[1:]:
+                task.delete()
+        raise RuntimeError(f'_request_match_task already exists for {slack_team_id}')
```

Comment:
Does an error really need to be logged and raised here? Why not just delete extras silently and return?

Reply:
you make good sense